### PR TITLE
Update status of units to mature.

### PR DESCRIPTION
**Pull request by mhvk**
What was holding it back was the idea that quantities would start to have associated uncertainties, but this has now moved to the new `uncertainty` sub-module.

p.s. I perhaps should have, but didn't check the status of all other submodules carefully.

@adrn - do you agree units can be transitioned from stable to mature?

@bsipocz - I think it would make sense to have this in 3.1 still...

### CODE REVIEW

File: docs/stability.rst [hunk 1]
```diff
@@ -257,10 +257,10 @@ The current planned and existing sub-packages are:
                 astropy.units
             </td>
             <td align='center'>
-                <span class="stable"></span>
+                <span class="mature"></span>
             </td>
             <td>
-                New in v0.2. Current functionality stable with intent to maintain backwards compatibility. Significant new functionality, in particular to allow dealing with uncertainties, is likely to be added in future versions.
+                Incremental improvements since v0.4. Functionality mature and unlikely to change. Efforts focussed on performance and increased interoperability with numpy functions.
```
Comment (pllim) [inline docs/stability.rst]: Nitpick: Capitalize "numpy"

Typo: "focused"

File: docs/stability.rst [hunk 2]
```diff
@@ -257,10 +257,10 @@ The current planned and existing sub-packages are:
                 astropy.units
             </td>
             <td align='center'>
-                <span class="stable"></span>
+                <span class="mature"></span>
             </td>
             <td>
-                New in v0.2. Current functionality stable with intent to maintain backwards compatibility. Significant new functionality, in particular to allow dealing with uncertainties, is likely to be added in future versions.
```
Comment (pllim) [inline docs/stability.rst]: So I guess the "uncertainties" thingy is not happening?

File: docs/stability.rst (hunk 1 above)
Comment (mhvk) [reply docs/stability.rst]: Done

File: docs/stability.rst (hunk 2 above)
Comment (mhvk) [reply docs/stability.rst]: It is - but it is its own submodule `uncertainty`, which is correctly marked as "in heavy development"

**Comment by bsipocz** [comment]
@mhvk - This should definitely land in 3.1. I'm waiting for one more PR for RC, but then the actual release will happen only after more testing.

**Comment by bsipocz** [comment]
Thanks @mhvk!

**Pull request merged**